plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergAnalyze.java

```diff
@@ -0,0 +1,377 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static io.trino.plugin.iceberg.IcebergQueryRunner.createIcebergQueryRunner;
+import static io.trino.testing.TestingAccessControlManager.TestingPrivilegeType.EXECUTE_TABLE_PROCEDURE;
+import static io.trino.testing.TestingAccessControlManager.privilege;
+import static java.lang.String.format;
+
+public class TestIcebergAnalyze
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createIcebergQueryRunner();
+    }
+
+    @Test
+    public void testAnalyze()
+    {
+        String tableName = "test_analyze";
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.nation", 25);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, null, 0, null, '0', '24')," +
+                        "('regionkey', null, null, 0, null, '0', '4')," +
+                        "('comment', null, null, 0, null, null, null)," +
+                        "('name', null, null, 0, null, null, null)," +
+                        "(null, null, null, null, 25, null, null)");
+
+        assertUpdate("ANALYZE " + tableName);
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, 25, 0, null, '0', '24')," +
+                        "('regionkey', null, 5, 0, null, '0', '4')," +
+                        "('comment', null, 25, 0, null, null, null)," +
+                        "('name', null, 25, 0, null, null, null)," +
+                        "(null, null, null, null, 25, null, null)");
+
+        // reanalyze data
+        assertUpdate("ANALYZE " + tableName);
```

Comment:
Could you add a test at the Iceberg table level to retrieve the statistics files for the iceberg table and showcase that calling `ANALYZE` `2` times on the same table snapshot creates `??` copies of the stats?

Reply:
adding `testReanalyze`